### DISCUSSION TITLE
Don't minify framework when building?

### DISCUSSION THREAD

**Discussion by vegerot** [vercel/next.js#57292] [Help]
### Summary

I am trying to **profile the performance** of my website.  In order to more easily read the performance traces, I'd like to use **unminified** versions of both my app code and the **Next.js framework code** in Production builds.

✅ I was able to disable minification in my app code by setting `config.webpack.optimization.minimize` to false.

❌ **Disabling minification in the framework code** has been more complicated.  Here's what I've done so far:

1. Cloned Next.js
1. followed [Developing](https://github.com/vercel/next.js/blob/canary/contributing/core/developing.md) to get Next running locally
2. Checked out `v12.3.4` which is the version my website uses
3. `pnpm dev`
4. followed [Developing Using Your Local Version of Next.js](https://github.com/vercel/next.js/blob/canary/contributing/core/developing-using-local-app.md#developing-using-your-local-version-of-nextjs)

Expected:
Should show unminified function names

Actual:
Website works, but still uses minified code.  I can see that in `dist/apps/journey/app/.next/static/chunks/framework-0e49fded8df17ef1.js` it is still minifying function names.

I can see in the Next repo that it's not being minified, so the minification must be happening in my app even though I disabled minification.  Would someone please help me link Next without minifying the code (or at least function/variable names)?  Alternatively, **showing source maps for the Next/React internals** would be great. Ideally I'd like to not minify Next/React (to make debugging easier), but sourcemaps would be fine.

### Additional information

Next.js version: **12.3.4**

```js
❯ pnpm next info

    Operating System:
      Platform: linux
      Arch: x64
      Version: #44~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Mon May 22 13:39:36 UTC 2
    Binaries:
      Node: 16.10.0
      npm: 7.24.0
      Yarn: 1.22.19
      pnpm: 7.25.0
    Relevant packages:
      next: 12.3.4
      eslint-config-next: 12.1.5
      react: 17.0.2
      react-dom: 17.0.2

warn  - Latest canary version not detected, detected: "12.3.4", newest: "13.5.7-canary.17".
        Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
        Read more - https://nextjs.org/docs/messages/opening-an-issue
```

Please see [related StackOverflow question](https://stackoverflow.com/questions/77348421/how-to-profile-next-js-app-using-unminified-function-names) if people have responded there.

**Comment by icyJoseph** [comment]
This is trickier than it seems, because most frameworks use some kind of "signal" to use their production mode. There are at least two things to think about here.

First, that setting this "signal" impacts code globally, for example `process.env.NODE_ENV` set to development or production, has massive implications on most code bases.

Second, that this implications are not just minification, but might imply entirely different code to be loaded. For example a library might require different modules for production and development. 

- https://overreacted.io/how-does-the-development-mode-work/

So, to get this done, there's no straightforward way... you'd need to set `NODE_ENV` to development, under the umbrella of `next build` and `next start`, or rather, have those under the umbrella of `NODE_ENV` development, but that might load code that cannot even work outside of a development environment.

**Reply by vegerot** [reply]
Sorry, I should have clarified my question better:
I want to use the unminified **production** code.  Development mode makes various changes to the functionality of your code, as you mentioned "implications are not just minification, but might imply entirely different code to be loaded".  I want to run the production version of Next and React, but to have the code unminified (or at least with sourcemaps) so I can profile it more easily.

**Reply by vegerot** [reply]
For example, for `react-dom` there are two bundles: `react-dom.development.js` and `react-dom.production.min.js`.  I want to use `react-dom.production.js`, but I don't know how to generate and link it.
